lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -334,31 +458,55 @@ def __init__(self, config: PbenchServerConfig, logger: Logger, schema: Schema):
         port = config.get("elasticsearch", "port")
         self.es_url = f"http://{host}:{port}"
         self.schema = schema
+        self.role = role
 
-    def _get_user_term(self, user: str) -> dict:
+    def _get_user_term(self, json: JSON) -> Dict[str, str]:
         """
-        _get_user_term Generate the user term for an Elasticsearch document
-        query. If the specified user parameter is not None, search for
-        documents with that value as the authorized owner. If the user
-        parameter is None, instead search for documents with public access.
+        Generate the user term for an Elasticsearch document query. If the
+        specified "user" parameter is not None, search for documents with that
+        value as the authorized owner. If the "user" parameter is absent or
+        None, and the "access" parameter is "public", search for all published
+        documents.
+
+        TODO: Currently this code supports either all published data, or all
+        data owned by a specified user. More work in query construction must
+        be completed to support additional combinations. I plan to support
+        additional flexibility in a separate PR based on issue #2370.
 
         Args:
-            user: Pbench internal username or None
+            JSON query parameters containing keys:
+                "user": Pbench internal username if present and not None
+                "access": Access category, "public" or "private"; defaults
+                    to "private" if "user" is specified, and "public" if
+                    user is not specified. (NOTE: other combinations are
+                    not currently supported.)
+
+        Raises:
+            UnsupportedAccessMode: This is a temporary situation due to the
+            current query limitations. When the full semantics are complete
+            there will be no "illegal" constraint combinations but only
+            combinations that yield no data (e.g, "private" data for another
+            user).
 
         Returns:
             An assembled Elasticsearch query term
         """
+        user = json.get("user")
+        access = json.get(
+            "access", Dataset.PRIVATE_ACCESS if user else Dataset.PUBLIC_ACCESS
+        )
         if user:
             term = {"authorization.owner": user}
+        elif access == Dataset.PRIVATE_ACCESS:
+            raise UnsupportedAccessMode(user, access)
         else:
-            term = {"authorization.access": "public"}
+            term = {"authorization.access": access}
```

Comment:
Yes; your sample code still uses `access` in the `raise`, but of course at that point we know that it's `Dataset.PRIVATE_ACCESS` since we just tested it.

Reply:
> isn't the access value always gonna be public?
@npalaska, "always" is a very long time!  😁 All we will know is that the `access` value is _not_ "private" -- we cannot state that it _is_ "public" unless and until we check that there aren't any other values.  At the moment, there aren't any, but that could easily change without this code knowing about it, so it would be best not to code in assumptions like that.

Reply:
Ha; I knew you'd weigh in on that. However, since this entire method needs revision to deal with the more flexible mechanism we've proposed (even without generalized groups), I was inclined to accept Nikhil's suggestion that, within the current limitations, I'd over-complicated it.

When I get this PR merged, I'll start breaking it.

Reply:
😁 
I'm unconvinced that it's overcomplicated (and, I don't see a reason to change it).
But, yeah, let's break it later.  😉

Reply:
OK, deferring the `json.get("access")` call was worth changing.  (Good eye, @npalaska!)